### DISCUSSION TITLE
Can `TimeBasedRollingPolicy` and `SizeBasedTriggeringPolicy` be used at the same time?

### DISCUSSION THREAD

**Discussion by SageJustus** [qos-ch/logback#846] [Q&A]
I found a [logback tutorial](https://stackify.com/logging-logback/) with this code:

> To better understand these policies, let’s create an appender which makes use of a TimeBasedRollingPolicy and a SizeTriggeringPolicy:
```xml
<appender name="rollingFile" class="ch.qos.logback.core.rolling.RollingFileAppender">
    <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
        <fileNamePattern>log-%d{yyyy-MM-dd}.log</fileNamePattern>
        <maxHistory>30</maxHistory> 
        <totalSizeCap>3GB</totalSizeCap>
    </rollingPolicy>
    <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
        <maxFileSize>3MB</maxFileSize>
    </triggeringPolicy>
    <encoder>
        <pattern>[%thread] %-5level %logger{36} - %msg%n</pattern>
    </encoder>
</appender>
```

> The TimeBasedRollingPolicy implements both a RollingPolicy and a TriggeringPolicy.
> 
> The example above configures the fileNamePattern attribute based on the day – which means the name of each file contains the current date, and also that the rollover will happen daily.
>
> Notice how we’re limiting the log data here – maxHistory is set to a value of 30, alongside a totalSizeCap of 3GB – which means that the archived logs will be kept for the past 30 days, up to a maximum size of 3GB.
> 
> Finally, the SizeBasedTriggeringPolicy defined configures the rollover of the file whenever it reaches 3 MB. Of course that’s quite a low limit...

But when I refer to this code and set `maxFileSize` to `100KB`, I find that only `TimeBasedRollingPolicy` is working and `SizeBasedTriggeringPolicy` is not.

After that I found a `log4j` [answer](https://stackoverflow.com/a/8669709/14887783).
> Hence, your `SizeBasedTriggeringPolicy` gets ignored since `TimeBasedRollingPolicy` is configured above. 

So, Can `TimeBasedRollingPolicy` and `SizeBasedTriggeringPolicy` be used at the same time?

I also asked a question on [Stack Overflow](https://stackoverflow.com/q/78888650/14887783)

**Comment by ceki** [comment]
@SageJustus  Please refer to the documentation at https://logback.qos.ch/manual/appenders.html#SizeAndTimeBasedRollingPolicy

**Reply by SageJustus** [reply]
I know `SizeAndTimeBasedRollingPolicy`, but I want know is it effective to use TimeBasedRollingPolicy and SizeBasedTriggeringPolicy at the same time?
